### PR TITLE
VEN-1188 | Give customer a berth without application

### PR DESCRIPTION
**Pull request by TatuArvela**
## Description :sparkles:

Admin needs to give customer a berth without an application

* **Click Lisää uusi from berths section**
* **Choose harbour from harbour list**
* Choose berth (not yet implemented)
* Preview invoice/offer in single customer view (not yet implemented)
* Send invoice from single customer view (not yet implemented)

I implemented this with an experimental hook + component approach to avoid making CustomerView any more complex than it already is. If this approach is suitable, we can refactor other features to the same style and cut down the complexity.

Other improvements:

* Refactored `LeasesCard`

### Closes :no_good_woman:

* [VEN-1188](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1188): FE: Admin needs to give customer a berth without an application

## Screenshots :camera_flash:
image

### CODE REVIEW

**Comment by Nizar-Rahme** [comment]
@TatuArvela it's ready for review

File: src/features/customerView/CustomerViewContainer.tsx [hunk 1]
```diff
@@ -220,6 +223,7 @@ const CustomerViewContainer = () => {
         />
       </Modal>
 
+      {/* TODO: Send invoices for berth leases */}
```
Comment (TatuArvela) [inline src/features/customerView/CustomerViewContainer.tsx:226]: Is this TODO still valid?

File: src/features/customerView/boatsCard/__tests__/__snapshots__/BoatsCard.test.tsx.snap [hunk 2]
```diff
@@ -1,3 +1,3 @@
...
...um</span></div><div class=\\"labelValuePair\\"><span class=\\"label standard\\">Käyttötarkoitus:</span><span class=\\"value left\\">For fun</span></div></section></article><article class=\\"section\\"><section class=\\"body\\"><div class=\\"boatCertificate\\"><div class=\\"boatCertificateLabel\\">Katsastustodistus:</div><div class=\\"boatCertificateValues\\"><div><span class=\\"text critical span\\">Tiedosto puuttuu</span></div><div><span class=\\"text standard span\\">Voimassa 04.09.2099 asti</span></div><div>Tarkastettu 28.04.2020</div><div>Jukka Virtanen</div></div></div></section></article><article class=\\"section\\"><section class=\\"body\\"><div class=\\"boatCertificate\\"><div class=\\"boatCertificateLabel\\">Vakuutuskirjaote:</div><div class=\\"boatCertificateValues\\"><div><a class=\\"link default\\" href=\\"https://venepaikka-api.test.hel.ninja/media/boats/2c736de2-5659-46ea-ad3f-9c1364307d7d/dummy_pdf2.pdf\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">dummy_pdf2.pdf</a></div><div>Tarkastettu 28.04.2020</div></div></div></section></article></div><div class=\\"cardBody createBoat\\"><button><span class=\\"text brand span\\">Lisää uusi</span></button></div></div>"`;
```
Comment (TatuArvela) [inline src/features/customerView/boatsCard/__tests__/__snapshots__/BoatsCard.test.tsx.snap:3]: This snapshot could be switched to `.render()`

File: src/features/customerView/createLease/createLeaseModal.module.scss [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  .select {
+    width: 100%;
+    margin-bottom: units(2);
+  }
+
+  .buttons {
+    display: flex;
+    justify-content: space-between;
+    // margin-top: units(1);
```
Comment (TatuArvela) [inline src/features/customerView/createLease/createLeaseModal.module.scss:19]: This commented line should probably be removed

File: src/features/customerView/createLease/useCreateLease.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import React, { useState } from 'react';
+
+import CreateLeaseModal from './CreateLeaseModal';
+
+// FIXME: This comment is intended for whoever picks this up next week
+// This approach is an attempt to clear up the amount of state, query and render control in CustomerView,
+// by separating modal features into a hook and a component. Feel free to use this or change it as you see fit!
+// With an approach like this, this whole flow can be attached to any view. Perhaps this could be under 'features'?
```
Comment (TatuArvela) [inline src/features/customerView/createLease/useCreateLease.tsx:8]: What about this comment?

**Review by TatuArvela — COMMENTED** [review]
👍  Seems clear, added a few comments

File: src/features/customerView/CustomerViewContainer.tsx (hunk 1 above)
Comment (Nizar-Rahme) [reply src/features/customerView/CustomerViewContainer.tsx:226]: Oh, I forgot about that 😓

File: src/features/customerView/boatsCard/__tests__/__snapshots__/BoatsCard.test.tsx.snap (hunk 2 above)
Comment (Nizar-Rahme) [reply src/features/customerView/boatsCard/__tests__/__snapshots__/BoatsCard.test.tsx.snap]: Changed it.

File: src/features/customerView/createLease/createLeaseModal.module.scss (hunk 3 above)
Comment (Nizar-Rahme) [reply src/features/customerView/createLease/createLeaseModal.module.scss]: Removed

File: src/features/customerView/createLease/useCreateLease.tsx (hunk 4 above)
Comment (Nizar-Rahme) [reply src/features/customerView/createLease/useCreateLease.tsx]: I had one issue with this approach that the data was queried when the render modal method was called; meaning when the customer page was loaded and I couldn't manage to make it fetch its own data separately whenever you open the modal.
This is a problem if you were to create a new boat for example, you had to refresh the while page.
Therefore, I switched the custom hook back to our conventional way.

Comment (TatuArvela) [reply src/features/customerView/createLease/useCreateLease.tsx]: 👍  Fair enough, I guess it was worth a try

File: src/features/invoiceCard/__tests__/InvoiceCard.test.tsx [hunk 5]
```diff
@@ -35,24 +34,24 @@ describe('InvoiceCard', () => {
   describe('send button', () => {
     const findSendButton = (wrapper: ReactWrapper) => wrapper.find('.buttonRow').find('div').find('button');
 
-    it('is not displayed when application status not "generated" or "sent"', () => {
-      Object.values(ApplicationStatus)
-        .filter((status) => status !== ApplicationStatus.OFFER_GENERATED && status !== ApplicationStatus.OFFER_SENT)
+    it('is not displayed when application status not "DRAFTED" or "OFFERED"', () => {
```
Comment (TatuArvela) [inline src/features/invoiceCard/__tests__/InvoiceCard.test.tsx:37]: application -> lease

**Review by TatuArvela — COMMENTED** [review]
🟢

**Review by Nizar-Rahme — APPROVED** [review]

**Pull request merged**